Sources/Bodega/StorageEngine.swift

```diff
@@ -86,4 +87,26 @@ extension StorageEngine {
         let allKeys = await self.allKeys()
         return await self.readDataAndKeys(keys: allKeys)
     }
+    
+    /// Checks whether a value with a key is persisted.
+    /// - Parameter key: The key to check for existence.
+    /// - Returns: If the key exists the function returns true, false if it does not.
+    public func keyExists(_ key: CacheKey) async -> Bool {
```

Comment:
I'm not sure if I'm for or against adding a default `keyExists`. The two bundled StorageEngines have very different implementations, and I have to imagine that this is something a StorageEngine author should spend time thinking about.

The argument for adding it is that it is a pretty reasonable default, but my argument against it is that the performance characteristics are pretty different. If I want to ensure a StorageEngine author considered the differences then I think it might be worth having them be forced to write out their own implementation.

What do you think @dannynorth?

Reply:
Since all the existing `StorageEngines` implement this, leaving off the default implementation is reasonable.

Reply:
Done ✅

Reply:
Perfect, thanks a lot!